python/ray/dag/tests/experimental/test_collective_dag.py

```diff
@@ -45,6 +47,60 @@ def compute_with_tuple_args(self, args, i: int):
         return tensor
 
 
+class AbstractNcclGroup(GPUCommunicator):
+    """
+    A dummy NCCL group for testing.
+    """
+
+    def __init__(self, actor_handles: List[ray.actor.ActorHandle]):
+        self._actor_handles = actor_handles
+        self._rank = None
+
+    def initialize(self, rank: int) -> None:
+        self._rank = rank
+
+    def get_rank(self, actor: ray.actor.ActorHandle) -> int:
+        return self._actor_handles.index(actor)
+
+    def get_world_size(self) -> int:
+        return len(self._actor_handles)
+
+    def get_self_rank(self) -> Optional[int]:
+        return self._rank
+
+    def get_actor_handles(self) -> List["ray.actor.ActorHandle"]:
+        return self._actor_handles
+
+    def send(self, value: "torch.Tensor", peer_rank: int) -> None:
+        raise NotImplementedError
+
+    def recv(
+        self,
+        shape: Tuple[int],
+        dtype: "torch.dtype",
+        peer_rank: int,
+        allocator: Optional[TorchTensorAllocator] = None,
+    ) -> "torch.Tensor":
+        raise NotImplementedError
+
+    def allreduce(
+        self,
+        send_buf: "torch.Tensor",
+        recv_buf: "torch.Tensor",
+        op: ReduceOp = ReduceOp.SUM,
+    ) -> None:
+        raise NotImplementedError
+
+    def destroy(self) -> None:
+        pass
+
+
+def mock_do_init_nccl_group(self, rank: int, nccl_group_id: str, custom_nccl_group: GPUCommunicator):
```

Comment:
Can you tell me what's the difference between this and the actual `do_init_nccl_group`? In the actual one, `custom_nccl_group` is optional, why it isn't here?

Reply:
Inside `MockNcclGroupSet`'s `mock_init`, if the NCCL group is None, we create an `AbstractNcclGroup` and initialize every rank with it.